### PR TITLE
Tests for `[[reqd_sub_group_size]]` attribute from optional kernel features 

### PR DESCRIPTION
**Pull request by andreyromanof**
This PR provides tests for the exception that are thrown by [[sycl::reqd_sub_group_size(N)]] attribute with unsupported N.

Changed files related to this PR: [link](https://github.com/KhronosGroup/SYCL-CTS/pull/289/files/b90a11e64f1688617ee632cb4285b16a908c8c63..HEAD)

Depends on:
- #287

### CODE REVIEW

File: tests/optional_kernel_features/kernel_features_sub_group_size_exceptions.cpp [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+  // Verify N supported or not supported as sub_group size on the current device
+  const auto sg_sizes_vec =
+      queue.get_device().get_info<sycl::info::device::sub_group_sizes>();
+  auto find_res = std::find(sg_sizes_vec.begin(), sg_sizes_vec.end(), N);
+  if (find_res != sg_sizes_vec.end()) {
+    is_exception_expected = false;
+  }
+
+  {
+    const auto separate_lambda_no_arg = []() [[sycl::reqd_sub_group_size(N)]]{};
```
Comment (keryell) [inline tests/optional_kernel_features/kernel_features_sub_group_size_exceptions.cpp:43]:
```suggestion
    const auto separate_lambda_no_arg = [] [[sycl::reqd_sub_group_size(N)]]{};
```

**Review by keryell — APPROVED** [review]
Thanks.

**Comment by andreyromanof** [comment]
@psalz 
Is it allowed to pass two implementations to `DISABLED_FOR_TEMPLATE_TEST_CASE_SIG`? I'm getting an [error](https://github.com/KhronosGroup/SYCL-CTS/runs/5355352978?check_suite_focus=true#step:8:15446) that says `too few arguments provided`.

**Comment by psalz** [comment]
> @psalz Is it allowed to pass two implementations to `DISABLED_FOR_TEMPLATE_TEST_CASE_SIG`? I'm getting an [error](https://github.com/KhronosGroup/SYCL-CTS/runs/5355352978?check_suite_focus=true#step:8:15446) that says `too few arguments provided`.

It should be, but I may have botched something during refactoring before I made that PR last week. I'm looking into it!

**Comment by psalz** [comment]
My bad, fixed in #295.

**Comment by andreyromanof** [comment]
Thanks

**Comment by bader** [comment]
@andreyromanof, please, resolve merge conflicts.

**Comment by andreyromanof** [comment]
> @andreyromanof, please, resolve merge conflicts.

Done.

**Pull request merged**